specification/saas/resource-manager/Microsoft.SaaS/preview/2018-03-01-beta/saas.json

```diff
@@ -0,0 +1,866 @@
+          "description": "The offer id."
+        },
+        "publisherId": {
+          "type": "string",
+          "description": "The publisher id."
+        },
+        "quantity": {
+          "type": "number",
+          "description": "The seat count."
+        },
+        "skuId": {
+          "type": "string",
+          "description": "The plan id."
+        },
+        "paymentChannelType": {
+          "type": "string",
+          "description": "The Payment channel for the SaasSubscription.",
+          "enum": [
+            "SubscriptionDelegated",
+            "CustomerDelegated"
+          ],
+          "x-ms-enum": {
+            "name": "paymentChannelType",
+            "modelAsString": true
+          }
+        },
+        "paymentChannelMetadata": {
+          "type": "object",
+          "description": "The metadata about the SaaS subscription such as the AzureSubscriptionId and ResourceUri.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "saasResourceName": {
+          "type": "string",
+          "description": "The SaaS resource name."
+        },
+        "termId": {
+          "type": "string",
+          "description": "The current Term id."
+        },
+        "autoRenew": {
+          "type": "boolean",
+          "description": "Whether the SaaS subscription will auto renew upon term end."
+        },
+        "publisherTestEnvironment": {
+          "type": "string",
+          "description": "The environment in the publisher side for this resource."
+        }
+      }
+    },
+    "AccessTokenResult": {
+      "description": "the ISV access token result response.",
+      "type": "object",
+      "properties": {
+        "publisherOfferBaseUri": {
+          "description": "The Publisher Offer Base Uri",
+          "type": "string"
+        },
+        "token": {
+          "description": "The generated token",
+          "type": "string"
+        }
+      }
+    },
+    "SaasAppResponseWithContinuation": {
+      "description": "saas app response with continuation.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "the next link to query to get the remaining results.",
+          "type": "string"
+        },
+        "value": {
+          "description": "the value of response.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SaasApp"
+          }
+        }
+      }
+    },
+    "SaasAppOperationsResponseWithContinuation": {
+      "description": "saas app operation response with continuation.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "the next link to query to get the remaining results.",
+          "type": "string"
+        },
+        "value": {
+          "description": "the value of response.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SaasAppOperation"
```

Comment:
>SaasAppOperation [](start = 35, length = 16)

Please take a look at [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json). You should be able to reuse the operation definition from that file. (Swagger allows you to reference content from another swagger file.)